cmd/ctl/pkg/create/certificaterequest.go

```diff
@@ -0,0 +1,287 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	restclient "k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	cmapiv1alpha2 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+	"github.com/jetstack/cert-manager/pkg/webhook"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+Create a cert-manager CertificateRequest resource and store private key on local file.`))
+
+	example = templates.Examples(i18n.T(`
+# Create a certificate request from from file.
+kubectl cert-manager create certificaterequest -f my-certificate.yaml
+
+# Create a certificate request in namespace default, provided no conflict with namespace defined in file.
+kubectl cert-manager create certificaterequest --namespace default -f my-certificate.yaml
+
+# Create a certificate request with the name 'my-cr', private key will be stored in file 'my-cr.key'.
+kubectl cert-manager create certificaterequest my-cr -f my-certificate.yaml
+
+# Create a certificate request and store private key in file 'new.key'.
+kubectl cert-manager create certificaterequest my-cr -f my-certificate.yaml --output-key-file new.key
+`))
+)
+
+var (
+	// Use the webhook's scheme as it already has the internal cert-manager types,
+	// and their conversion functions registered.
+	// In future we may we want to consider creating a dedicated scheme used by
+	// the ctl tool.
+	scheme = webhook.Scheme
+)
+
+// Options is a struct to support create certificaterequest command
+type Options struct {
+	CMClient         cmclient.Interface
+	RESTConfig       *restclient.Config
+	CmdNamespace     string
+	EnforceNamespace bool
+	KeyFileName      string
+
+	resource.FilenameOptions
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdCreateCR returns a cobra command for create CertificateRequest
+func NewCmdCreateCR(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:     "certificaterequest",
+		Aliases: []string{"cr"},
+		Short:   "Create a CertificateRequest resource",
+		Long:    long,
+		Example: example,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 1 {
+				return fmt.Errorf("only one argument can be passed in: the name of the certificate request")
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, "Path to the manifest of a Certificate resource")
```

Comment:
That's because we use the `AddFilenameOptionFlags` function, and it has a prefix already set. I'll make it more readable.
Btw, just to note and I don't forgetm convert has a similar help message.

Reply:
I changed `AddFilenameOptionFlags` to `AddJsonFilenameFlag` which only adds the -f flag, since right now we only support one single yaml input file anyway. And also that removes the prefix.